File: flutter/packages/gifts/lib/src/domain/entities/gift_category.dart
import 'package:equatable/equatable.dart';

/// A gift category (tab in the picker).
class GiftCategory extends Equatable {
  final int id;
  final String title;
  final String type;

  const GiftCategory({
    required this.id,
    required this.title,
    required this.type,
  });

  @override
  List<Object?> get props => [id];
}

File: backend/packages/gifts/src/Models/GiftCategory.php
<?php

namespace Utd\Gifts\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Support\Facades\Cache;

class GiftCategory extends Model
{
    protected $fillable = ['title', 'type', 'sort'];

    protected $casts = [
        'title' => 'array',
    ];

    /** Bust the cached catalog whenever a category changes (create/edit/reorder/delete). */
    protected static function booted(): void
    {
        $forget = fn () => Cache::forget('gifts:categories');
        static::saved($forget);
        static::deleted($forget);
    }

    public function gifts(): HasMany
    {
        return $this->hasMany(Gift::class);
    }
}
